haystack/utils/reflection.py

```diff
@@ -33,3 +37,47 @@ def wrapper_invocation_counter(*args, **kwargs):
 
     wrapper_invocation_counter.counter = 0
     return wrapper_invocation_counter
+
+
+def retry_with_exponential_backoff(
+    initial_delay: float = 1,
+    exponential_base: float = 2,
+    jitter: bool = True,
+    max_retries: int = 10,
+    errors: tuple = (OpenAIRateLimitError,),
+):
+    """Retry a function with exponential backoff."""
```

Comment:
Could you leave a docstring or a note explaining what units the delay is measured in? Right now it's unclear to me if it is in seconds or milliseconds.

Reply:
Right, good point